src/main/java/uk/gov/moj/sdt/service/FeedbackService.java

```diff
@@ -0,0 +1,133 @@
+
+    private IIndividualRequestDao individualRequestDao;
+
+    private XmlConverter xmlToObject;
+
+    public FeedbackService(@Qualifier("IndividualRequestDao")
+                               IIndividualRequestDao individualRequestDao,
+                           XmlConverter xmlToObject) {
+        this.individualRequestDao = individualRequestDao;
+        this.xmlToObject = xmlToObject;
+    }
+
+    public void feedback(String sdtRequestId, CMCFeedback cmcFeedback) {
+        IIndividualRequest individualRequest = individualRequestDao.getRequestBySdtReference(sdtRequestId);
+        String requestType = individualRequest.getRequestType();
+        if (RequestType.JUDGMENT.getType().equals(requestType)) {
+            updateJudgementRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.BREATHING_SPACE.getType().equals(requestType)) {
+            updateBreathingSpaceRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.CLAIM_STATUS_UPDATE.getType().equals(requestType)) {
+            updateClaimStatusUpdateRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.CLAIM.getType().equals(requestType)) {
+            updateCreateClaimRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.WARRANT.getType().equals(requestType)) {
+            updateWarrantRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.JUDGMENT_WARRANT.getType().equals(requestType)) {
+            updateCombinedJudgementWarrantRequest(cmcFeedback, individualRequest);
+        }
+        individualRequestDao.persist(individualRequest);
+    }
+
+    private void updateCreateClaimRequest(CMCFeedback cmcFeedback,
+                                          IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        ClaimResponse response = new ClaimResponse();
+        response.setClaimNumber(cmcFeedback.getClaimNumber());
+        response.setIssueDate(cmcFeedback.getIssueDate());
+        response.setServiceDate(cmcFeedback.getServiceDate());
+        individualRequest.setTargetApplicationResponse(convertObjectToXml(response));
+    }
+
+    private void updateJudgementRequest(CMCFeedback cmcFeedback,
+                                          IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        JudgementResponse response = new JudgementResponse();
+        response.setJudgmentEnteredDate(cmcFeedback.getJudgmentEnteredDate());
+        response.setFirstPaymentDate(cmcFeedback.getFirstPaymentDate());
+        individualRequest.setTargetApplicationResponse(convertObjectToXml(response));
+    }
+
+    private void updateWarrantRequest(CMCFeedback cmcFeedback,
+                                        IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        WarrantResponse response = new WarrantResponse();
```

Comment:
Should this also set the issueDate and judgementWarrantStatus on the WarrantResponse?  The issueDate is mentioned on the CMC Asynchronous Responses Confluence page.  The judgementWarrantStatus isn't, but the equivalent responses from MCOL will contain it.

Reply:
done